cli/azd/pkg/ext/hooks.go

```diff
@@ -0,0 +1,379 @@
+			matchingScripts = append(matchingScripts, scriptConfig)
+		}
+	}
+
+	return matchingScripts
+}
+
+func (h *CommandHooks) getImplicitHooks(prefix HookType, validHookNames []string) []*ScriptConfig {
+	matchingScripts := []*ScriptConfig{}
+
+	hooksDir := filepath.Join(h.cwd, ".azure", "hooks")
+	files, err := os.ReadDir(hooksDir)
+	if err != nil {
+		return matchingScripts
+	}
+
+	// Find implicit / convention based hooks in the .azure/hooks directory
+	// Find `predeploy.sh` or similar matching a hook prefix & valid command name
+	for _, file := range files {
+		fileName := file.Name()
+		fileNameWithoutExt := strings.TrimSuffix(fileName, path.Ext(fileName))
+
+		index := slices.IndexFunc(validHookNames, func(hookName string) bool {
+			return !file.IsDir() && hookName == fileNameWithoutExt
+		})
+
+		if index > -1 {
+			scriptType, err := inferScriptTypeFromFilePath(fileName)
+			if err != nil {
+				log.Printf("Found script hook '%s', but type is not supported. %s\n", fileName, err.Error())
+				continue
+			}
+
+			relativePath, err := filepath.Rel(h.cwd, filepath.Join(hooksDir, fileName))
+			if err != nil {
+				// This err should never happen since we are only looking for files inside the specified cwd.
+				log.Printf("Found script hook '%s', but is outside of project folder. Error: %s\n", file.Name(), err.Error())
+				continue
+			}
+
+			scriptConfig := ScriptConfig{
+				Name:     fileNameWithoutExt,
+				Path:     relativePath,
+				Location: ScriptLocationPath,
+				Type:     scriptType,
+			}
+
+			matchingScripts = append(matchingScripts, &scriptConfig)
+		}
+	}
+
+	return matchingScripts
+}
+
+func (h *CommandHooks) execScript(ctx context.Context, scriptConfig *ScriptConfig) error {
+	// Delete any temporary inline scripts after execution
+	defer func() {
+		if scriptConfig.Location == ScriptLocationInline {
+			os.Remove(scriptConfig.Path)
+		}
+	}()
+
+	script, err := h.GetScript(scriptConfig)
+	if err != nil {
+		return err
+	}
+
+	formatter := h.console.GetFormatter()
+	consoleInteractive := formatter == nil || formatter.Kind() == output.NoneFormat
+	scriptInteractive := consoleInteractive && scriptConfig.Interactive
+
+	// When running in an interactive terminal broadcast a message to the dev to remind them that custom hooks are running.
+	if consoleInteractive {
+		h.console.Message(
+			ctx,
+			output.WithBold(
+				fmt.Sprintf(
+					"Executing %s command hook => %s",
+					output.WithHighLightFormat(scriptConfig.Name),
+					output.WithHighLightFormat(scriptConfig.Path),
+				),
+			),
+		)
+	}
+
+	log.Printf("Executing script '%s'\n", scriptConfig.Path)
+	_, err = script.Execute(ctx, scriptConfig.Path, scriptInteractive)
+	if err != nil {
+		execErr := fmt.Errorf("failed executing script '%s' : %w", scriptConfig.Path, err)
+
+		// If an error occurred log the failure but continue
+		if scriptConfig.ContinueOnError {
+			h.console.Message(ctx, output.WithBold(output.WithWarningFormat("WARNING: %s", execErr.Error())))
```

Comment:
Is this error message user-friendly enough? I could see the most ideal version being "WARNING: 'postdeploy' hook failed with exit code: %s. Execution will continue since ContinueOnError is set to true".